compiler-lib/src/ast.rs

```diff
@@ -48,7 +48,7 @@ pub enum ClassMemberKind<'t> {
         Spanned<'t, ParameterList<'t>>,
         Spanned<'t, Block<'t>>,
     ),
-    MainMethod(Symbol, Spanned<'t, Block<'t>>),
+    MainMethod(Spanned<'t, ParameterList<'t>>, Spanned<'t, Block<'t>>),
```

Comment:
Why? The reason I made this a `Symbol` was to avoid  having to add a documented invariant stating that the list is "guaranteed to only contain [...]". Below, `method_params` should return `None` for `MainMethod`, since for all our intents and purposes the MainMethod has no parameter. It just has a symbol that must not be used.

Reply:
The only reason why this would make sense is that we have a span around the whole parameter. It is already enforced by the parser that the type of the argument is `String[]` and that the `MainMethod` only has one parameter.

Reply:
We need `args` for the local variable declaration scope in when doing name analysis. AFAIK it must not be shadowed by the body of the `main()` function unless main opens a new scope. but the body of `main` and the param list are in the same scope.